### ISSUE TITLE
Feature request: add field name as string

### ISSUE THREAD

**Issue by chu121su12** [skinny85/jilt#23]
Interoperating class fields in map (map's key, mostly json), I'd need the field name to be autogenerated as string. Granted this is not a builder feature, but it is an improvement in the generated class.

Basically, changing the current autogenerated class from:
```java
public class ObjBuilder {
  private Object field;
  public ObjBuilder field(final Object field) {
    this.field = field;
    return this;
  }
  //...
}
```
to:
```java
public class ObjBuilder {
  private Object _field;
  public ObjBuilder field(final Object field) {
    this._field = field;
    return this;
  }
  //...
  public static final String field = "field";
}
```

Is this acceptable to be added in?

**Comment by skinny85** [comment]
Hi @chu121su12,

thanks for opening the feature request.

Can you explain what does adding these public String fields accomplish? What's the goal of having them?

Thanks,
Adam

**Comment by chu121su12** [comment]
I'm sorry, the added field should be static (I've updated the issue code block).

Mostly this will be useful in working with map/string-based key. Say in simple http server you want to have filtering, such that:

```java
record FilterRequest(int age) {};

public void handle(FilterRequest rq, Server s)
{
  s.reply(
    Repository.filter(
      Map.ofEntries(Map.entry(FilterRequest.age, rq.age))
      // instead of 
      // Map.ofEntries(Map.entry("age", rq.age))
    )
  );
}
```

`FilterRequest.age` will still resolve to `"age"` but with class-bound definition, instead of a stray string.

Again, I realize this is not a builder-specific feature, just a DX feature.

**Comment by skinny85** [comment]
Right, but it wouldn't be `FilterRequest.age`, because Jilt can't modify existing classes/records, but rather `FilterRequestBuilder.age`, which I'm not sure is as appealing 🙂.

Let's keep this feature request open, and see if we get some +1s for it. Right now, this seems a little too niche for my taste 🙂.

**Comment by chu121su12** [comment]
Now that I have been using this more, I think I may need a more focused request, which is to add something like `memberPrefix` option. The addition of static member may be outside of the scope of the library, but if the private member names are adjustable, the static member can be added without changing much of the rest of the source.

**Comment by skinny85** [comment]
I don’t follow, sorry 🙂.

The static member can be added where?

**Comment by skinny85** [comment]
I'm closing this one due to inactivity. If you still need anything related to this issue, please comment, and I'll be happy to re-open.
